src/oidcc_auth_util.erl

```diff
@@ -0,0 +1,253 @@
+%%%-------------------------------------------------------------------
+%% @doc Authentication Utilities
+%% @end
+%%%-------------------------------------------------------------------
+-module(oidcc_auth_util).
+
+-feature(maybe_expr, enable).
+
+-include("oidcc_client_context.hrl").
+-include("oidcc_provider_configuration.hrl").
+
+-include_lib("jose/include/jose_jwk.hrl").
+
+-export_type([auth_method/0, error/0]).
+
+-type auth_method() ::
+    none | client_secret_basic | client_secret_post | client_secret_jwt | private_key_jwt.
+
+-type error() :: no_supported_auth_method.
+
+-export([add_client_authentication/6]).
+
+-spec add_client_authentication(
```

Comment:
Can you add `@private` to this function? I don't want this function to be part of the public interface.